subscriptions/subscriptions-impl/src/main/java/com/duckduckgo/subscriptions/impl/messaging/ItrMessagingInterface.kt

```diff
@@ -61,11 +61,11 @@ class ItrMessagingInterface @Inject constructor(
         try {
             val adapter = moshi.adapter(JsMessage::class.java)
             val jsMessage = adapter.fromJson(message)
-            val domain = runBlocking(dispatcherProvider.main()) {
-                webView.url?.toUri()?.host
+            val url = runBlocking(dispatcherProvider.main()) {
```

Comment:
Since we're using the URI from the webview, I'd suggest keeping the conversion to URI and not passing strings along the way. That way, `String.toTldPlusOne()` doesn't need to rely on `UriString.host(this)`, which is somewhat expensive, and could just do this.host instead. Probably not super relevant in this case, but I'm trying to remove String URIs when possible as part of the performance initiative, as the method I mentioned before introduces a significant toll when called a lot of times

Reply:
Makes sense. Changed to use the url string directly.

Reply:
FYI I'm going back to this here because the host is needed to calculate the eTLD+1 otherwise the result includes the path too. What's not needed is the reparsing to URI inside the toTldPlusOne method